### ISSUE TITLE
"Orphaned" blocks run in Blockly

### ISSUE THREAD

**Issue by motib** [Mobsya/blockly#13]
Students often leave orphaned blocks or scripts that are not enclosed in an event. Scratch simply ignores these until you click on them. Blockly runs the orphaned scripts are this is confusing. They should be ignored.

**Comment by stephanemagnenat** [comment]
I agree. Actually ideally they should be highlighted in red, and the program should not be able to run. These changes are not trivial because we embed a third-party program, rather than having full control as with VPL.

**Comment by motib** [comment]
We found this to be _extremely common_ in Scratch:
O. Meerbaum-Salant, M. Armoni, M. Ben-Ari. Habits of programming in Scratch. Sixteenth Conference on Innovation and Technology in Computer Science Education, Darmstadt, Germany, 2011, 168--172. 
That article condemns first dragging lots of blocks and then putting them together, not worrying about unused blocks.

Later a colleague from the CS department supported that method as close to a new programming paradigm they developed!
Michal Gordon, Assaf Marron, Orni Meerbaum-Salant. Spaghetti for the main course?: observations on the naturalness of scenario-based programming. Proceedings of the 17th ACM annual conference on Innovation and technology in computer science education. 2012, 198-203

I suggest being consistent with Scratch and simply ignoring scripts that are not within an event-handler.

**Comment by stephanemagnenat** [comment]
> I suggest being consistent with Scratch and simply ignoring scripts that are not within an event-handler.

I'm fine with that.
